packages/gatsby/src/redux/reducers/queries.ts

```diff
@@ -0,0 +1,236 @@
+import {
+  ActionsUnion,
+  IComponentState,
+  IGatsbyState,
+  IQueryState,
+} from "../types"
+
+type QueryId = string // page query path or static query id
+type ComponentPath = string
+type NodeId = string
+type ConnectionName = string
+
+export const FLAG_DIRTY_PAGE = 0b0001
+export const FLAG_DIRTY_TEXT = 0b0010
+export const FLAG_DIRTY_DATA = 0b0100
+
+export const FLAG_ERROR_BABEL = 0b0001
```

Comment:
NIT: should we call this `FLAG_ERROR_EXTRACTION`? From what I understand the reason (babel/graphql) doesn't matter so much and this flag is used both for babel errors (so invalid js syntaxes) and graphql errors (so invalid graphql query syntaxes)

Reply:
Changed in https://github.com/gatsbyjs/gatsby/pull/27504/commits/f3c056fdcfeca195d3969d68ec83f88261c7606f